File: TrainingCshar/Class/Encoder/Codificacion.cs
using System;

namespace TrainingCshar.Encoder
{
    class Codificacion
    {
        private static string _cadena;

        public static string Cadena()
        {
            return _cadena;
        }


        public static void Serializar(string cadena, int accion = 0)
        {
            switch (accion)
            {
                case 0:
                    _cadena = Encriptar(cadena);
                    break;
                case 1:
                    _cadena = Desencriptar(cadena);
                    break;
                default:
                    _cadena = cadena;
                    break;
            }
        }


        private static string Encriptar(string cadena)
        {
            byte[] encryted = System.Text.Encoding.Unicode.GetBytes(cadena);
            _cadena = Convert.ToBase64String(encryted);
            return _cadena;
        }


        private static string Desencriptar(string cadena)
        {
            byte[] decryted = Convert.FromBase64String(cadena);
            _cadena = System.Text.Encoding.Unicode.GetString(decryted);
            return _cadena;
        }

    }
}

File: TrainingCshar/Class/Encoder/Codificacion.vb
Namespace TrainingCshar.Encoder
    Friend Class Codificacion
        Private Shared _cadena As String

        Public Shared Function Cadena() As String
            Return _cadena
        End Function

        Public Shared Sub Procesar(ByVal cadena As String, ByVal codifica As Boolean)
            cadena = If(codifica, Encriptar(cadena), Desencriptar(cadena))
        End Sub

        Private Shared Function Encriptar(ByVal cadena As String) As String
            Dim encriptado = Encoding.Unicode.GetBytes(cadena)
            _cadena = Convert.ToBase64String(encriptado)
            Return _cadena
        End Function

        Private Shared Function Desencriptar(ByVal cadena As String) As String
            Dim desencriptado = Convert.FromBase64String(cadena)
            _cadena = Encoding.Unicode.GetString(desencriptado)
            Return _cadena
        End Function
    End Class
End Namespace
